### DISCUSSION TITLE
Is it possible to add GitHub Apps Bot to CODEOWNERS?

### DISCUSSION THREAD

**Discussion by 0gajun** [community/community#23064] [New to GitHub]
<p>Hi, there</p>
<p>I wanna add my GitHub Apps bot to CODEOWNERS to automate approval processes.</p>
<p>I’ve already tried to do it like following, but not work.</p>
<pre><code class="lang-auto">/example-path @our-githubapps-bot-name  
</code></pre>
<p>The bot has access to the repository with read&amp;write permission.</p>
<p>Is it possible to add the bot to CODEOWNERS or not? </p>
<p>If possible, how can I define CODEOWNERS?</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/0gajun">@0gajun</a>,</p>
<p>Thank you for being here! Currently, GitHub Apps can’t be used in CODEOWNERS – that’s not supported. It’s something the team is considering for the future, and I’ll be sure to add your use case to the internal feature request.</p>

**Reply by cjthom03** [reply]
Hi, has there been any movement on this? Thanks.

**Reply by CatalinaMoisuc** [reply]
Any news on this? Is this planned?

We also need it to be able to add https://github.com/apps/renovate-approve as a code owner so we can auto-merge dependency updates PRs opened by Renovate.

@AndreaGriffiths11 is there a GitHub issue opened for this feature request? Would be cool to be able to subscribe to it for updates.

**Reply by Shegox** [reply]
Just to add the workaround we are using internally, might be helpful for somebody else as well:

> Due to a limitation in GitHub, it's not possible to assign an GitHub App, such as this one (`renovate-approver`), as a `CODEOWNER`. This means you cannot use `CODEOWNERS` directly with the Renovate-approver app.
> However, GitHub does allow you to effectively disable the `CODEOWNER` for specific files in your repository. You can do this by not assigning a `CODEOWNER` to those files. 
> 
> For example, if you wish to disable the `CODEOWNER` for your `package.json` and `package-lock.json` files, you can do so by excluding them from your `CODEOWNERS` file like this:
> 
> ```CODEOWNERS
> *              some-codeowner-user
> package.json
> package-lock.json
> ```
> 
> In this configuration, `some-codeowner-user` is assigned as the `CODEOWNER` for all files in the repository (`*`) except for `package.json` and `package-lock.json`, which have no assigned `CODEOWNER`.

**Reply by pexa-slee** [reply]
Legend!

**Comment by hardboiled** [comment]
<p>I wrote a github app to merge code automatically if it passes CI checks and meets certain criteria that would designate the PR as a minimal change. For more involved changes, we do want owners to approve. It works in my test repo, but the repo I installed it on eventually has a CODEOWNERS file. When the app attempts to merge, I get an error message as shown below, even though it has merge access. There doesn’t seem to be any way to override this message, and this response is completely undocumented. Between this and github actions, I’ve had multiple mishaps and re-writes that I’ve had to do that have resulted in around 1-2 weeks of lost time on this project, primarily due to documentation issues and frankly bugs on what the docs say github supports. It would be nice if this stuff would work as documented and if bots with merge access wouldn’t get error messages back saying they can’t merge. Even just a force parameter to override the error message would give me what I need, but that doesn’t appear to exist.</p>
<p># with merge bot (github app) token<br>
$ curl -X PUT -H “Authorization: Bearer $INSTALL_TOKEN” -H “Accept: application/vnd.github.machine-man-preview+json” -H ‘Content-Type: application/json’ ‘<a href="https://api.github.com/repos/my-repo/pulls/234/merge" target="_blank" rel="nofollow noopener">https://api.github.com/repos/my-repo/pulls/234/merge</a>’ <br>
-d ‘{<br>
“commit_title”: “test merge by manually calling github API”,<br>
“commit_message”: “Test commit”,<br>
“merge_method”: “merge”<br>
}’ | python -m ‘json.tool’<br>
{<br>
“message”: “Waiting on code owner review from &lt;owner&gt;”,<br>
“documentation_url”: “<a href="https://help.github.com/articles/about-protected-branches" target="_blank" rel="nofollow noopener">https://help.github.com/articles/about-protected-branches</a>”<br>
}</p>
<p># with user API token<br>
curl -X PUT -H “Authorization: Bearer $GITHUB_TOKEN” -H “Accept: application/vnd.github.machine-man-preview+json” -H ‘Content-Type: application/json’ ‘<a href="https://api.github.com/repos/my-repo/pulls/234/merge" target="_blank" rel="nofollow noopener">https://api.github.com/repos/my-repo/pulls/234/merge</a>’ -d ‘{ <br>
“commit_title”: “test merge by manually calling github API”,<br>
“commit_message”: “Test commit”,<br>
“merge_method”: “merge”<br>
}’ | python -m ‘json.tool’<br>
{<br>
“sha”: “&lt;sha&gt;”,<br>
“merged”: true,<br>
“message”: “Pull Request successfully merged”<br>
}</p>

**Comment by Jdmorrisett** [comment]
<p>We also have a bot that needs the ability to force an approval and merge PRs. Where can we track this issue/feature request? Is it in a roadmap, ticket, or repo issue somewhere?</p>
<p>Best Regards</p>

**Comment by kamigerami** [comment]
bump

**Comment by pantelis-karamolegkos** [comment]
Is it possible however to add `github-actions` bot to CODEOWNERS?

**Reply by lukasz-mitka** [reply]
no

**Comment by omrizi-tr** [comment]
It it possible to add a GitHub app to the CODEOWNERS file?

**Comment by filippo-possenti-version1** [comment]
I also would find this very useful. Github, you should really look into this.

**Comment by TomerFi** [comment]
Waiting for this...

**Comment by m1n9o** [comment]
Eager to see this feature....

**Comment by gsuess** [comment]
Or at least allow the bot to be added to a team.

**Comment by nealajpatel** [comment]
The ability to add a github app to a team would be extremely helpful for our use-case given we use CODEOWNERS

**Comment by yatharth2307** [comment]
We are also looking forward to it. This would be extremely helpful.

**Comment by ronballesteros** [comment]
We also are using github apps to auto create and approve PRs.  We also use CODEOWNERS on our repos and this would be very useful for us.

**Comment by saeedseyfi** [comment]
Hey @AndreaGriffiths11!
We are also developing a PR review assistant app at Spotify. Allowing the app as codeowner or adding it to a team would be super helpful.
Is there any update regarding this request?

**Comment by suryapandian** [comment]
Hello
GH actions works like a charm. 👏 
However, GH bot not being a codeowner hinders our automation process.

Our org is also working towards an auto approval mechanism for renovate PRs. This would be extremely helpful. Waiting for this 🙇‍♀️

**Comment by davidforslof** [comment]
Hi!
We are also looking into this, +1 from EQT Partners

**Comment by damianperera** [comment]
Come on folks, this issue is 3 years old - any updates @AndreaGriffiths11?

**Reply by msimons** [reply]
+1

**Comment by Eeebru** [comment]
We are still waiting for the implementation of this! [at]github-team

**Comment by yehonadav-feel** [comment]
+1

**Comment by akashdip2001** [comment]
you can achieve automated approval processes with GitHub Apps by configuring your GitHub Actions workflow or using the GitHub API in your custom scripts. Here's a general approach:

1. **GitHub Actions Workflow:**
   - Create a GitHub Actions workflow that runs on pull requests.
   - Use the GitHub API to programmatically request reviews or approve pull requests.

   Example workflow YAML:
   ```yaml
   name: Automated Approvals

   on:
     pull_request:
       types:
         - opened

   jobs:
     automate-approval:
       runs-on: ubuntu-latest

       steps:
         - name: Checkout repository
           uses: actions/checkout@v2

         - name: Approve pull request
           run: |
             # Use GitHub API to approve the pull request
             # Example: curl -X POST -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
             #   "https://api.github.com/repos/${{ github.repository }}/pulls/${{ github.event.pull_request.number }}/reviews" \
             #   -d '{"event": "APPROVE"}'
   ```

2. **GitHub API in Custom Scripts:**
   - If you prefer to use custom scripts outside of GitHub Actions, you can make API requests to the GitHub API using a personal access token with appropriate permissions.

   Example using cURL:
   ```bash
   TOKEN="YOUR_PERSONAL_ACCESS_TOKEN"
   PR_NUMBER="123"

   curl -X POST -H "Authorization: token $TOKEN" \
     "https://api.github.com/repos/your-org/your-repo/pulls/$PR_NUMBER/reviews" \
     -d '{"event": "APPROVE"}'
   ```

Make sure to replace placeholders like `YOUR_PERSONAL_ACCESS_TOKEN`, `your-org`, and `your-repo` with your actual values.

Always be cautious with personal access tokens, and ensure they have the necessary permissions. It's good practice to use GitHub Actions with the `GITHUB_TOKEN` secret whenever possible, as it's automatically generated and has the appropriate permissions for the repository. Remember to adapt your scripts based on your specific requirements and security considerations.

## Profile
![Profile Image](https://thumbnails.odycdn.com/optimize/s:200:0/quality:95/plain/https://thumbs.odycdn.com/bf58cf4d7ec52c62f02c52e16f150a5b.webp)
**AKASHDIP MAHAPATRA**
"everything is physics bro"  
E=mc²

|[Subscribe on YouTube](https://www.youtube.com/channel/UCxvmp634YDc41xCWOdvWqoQ?sub_confirmation=1) |
|----  |

**Reply by AndresPinerosZen** [reply]
This is so disrespectful.

**Comment by lukasz-mitka** [comment]
Created this as a request in correct category: https://github.com/orgs/community/discussions/108490

**Reply by suryapandian** [reply]
thank you, i hope someone from github takes a look at it

**Comment by piotr-kowenzowski-asi** [comment]
+1

**Comment by brettcurtis** [comment]
We want to add @coderabbitai to our code owners.

**Reply by marin-keeper** [reply]
Add your vote here pls https://github.com/orgs/community/discussions/108490

**Comment by asos-lewish** [comment]
My organisation would also love the ability to allow a GitHub App the ability to approve PRs where branch protection requires a review from CODEOWNERS. Please can we get an idea if this is on your roadmap or how we can track progress on this?

**Reply by marin-keeper** [reply]
Please help push https://github.com/orgs/community/discussions/108490

**Comment by harshita-gupta** [comment]
This is a huge limitation in Github Apps, and is preventing us from deprecating old service accounts and migrating them fully to Github Apps.

**Reply by marin-keeper** [reply]
Please add your vote https://github.com/orgs/community/discussions/108490

**Comment by chrislewis-fd** [comment]
Is there an ETA on this feature? Its over 4 years old. Does anyone have a solid workaround for this?

**Reply by marin-keeper** [reply]
Link to the feature request https://github.com/orgs/community/discussions/108490

**Comment by LautaroLobo12** [comment]
@AndreaGriffiths11 any updates on this? Seems like a common use case that's missing rn

**Reply by marin-keeper** [reply]
No updates yet, vote here please https://github.com/orgs/community/discussions/108490

**Comment by austenstone** [comment]
Wouldn't a required status check act as a workaround? Sure you can't require approval for specific files but you could write that logic into your GitHub app and send a successful status check if those files aren't modified.

**Reply by BakerNet** [reply]
Yes, you can do this as a workaround, but you lose glance value on your PR for other checks.  Say you run CI checks - it's nice as a non-bot reviewer to see at a glance the most recent commit is green.

**Comment by younsl** [comment]
Unfortunately, As of February 2025, this is still not supported.

When attempting to add a bot to the `.github/CODEOWNERS` file using any of these formats, you'll get an `Invalid owner` error:

```bash
# .github/CODEOWNERS
* app/github-actions
* @github-actions
* @github-actions[bot]

**Reply by marin-keeper** [reply]
Here's the feat request for you to vote https://github.com/orgs/community/discussions/108490

**Comment by Hotpirsch** [comment]
We have an automatic PR validation running with Github Actions and it would be very beneficial if we could have the bot as the CODEOWNER because otherwise it happens regularly that some hyperactive user approves and clicks the merge button before the validation has actually finished. If the github-actions bot would be the code owner then that temptation could be avoided. +1

**Comment by busy-bee-happy** [comment]
This issue is now five years old. Could we please have an update on whether this is taken into consideration at all? It starts to feel a bit like messages are just sent into the void. The workarounds of adding your dependency definitions to the codeowners exclusion list breaks code ownership policies, and personal access tokens where the bot impersonates a real person introduces operational overhead (now token management, offboarding impersonated users). As such, developers refrain from automating dependency management, which introduces risk. Please help us automaters to automate our work, and we'll have more time to automate for the rest of the world. 

Thanks very much in advance <3

**Reply by marin-keeper** [reply]
https://github.com/orgs/community/discussions/108490 is the feature request, please add your vote

**Comment by m-perrott** [comment]
June 2025, we need an update on this.

**Reply by marin-keeper** [reply]
Feature request: https://github.com/orgs/community/discussions/108490
Add your vote please :)

**Comment by chlos** [comment]
We are using Atlantis/Terraform for some of our repos, and we wanted to add Atlantis bot to `CODEOWNERS` to allow it to merge PRs which met some special requirements. Basically we wanted Atlantis to run some custom policy checks / custom apply scripts to make it's own decision to merge some PRs ingoring the "Require review from Code Owners" branch protection rule.

Apparently adding the bot to the owners list is impossible (yet?). So we found this workaround useful:
* Use Rulesets instead of the Branch protection rules for your repo.
* Add the app to the bypass list.
* Now the app is able to merge PRs ignoring the `CODEOWNERS` settings.
<img width="1100" alt="Screenshot 2025-06-19 at 15 49 00" src="https://github.com/user-attachments/assets/af9eb9b1-dc73-4e6b-9e8d-c64a4ba8ec2e" />

**Comment by anthony-legay** [comment]
For those who still wonder, there seem to be a new setting possible now, I don't remember having it previously: "Exempt from rules" 
<img width="653" height="256" alt="image" src="https://github.com/user-attachments/assets/0b924147-5af8-4769-8509-d5a0e060527c" />
Description seems to fit the requirement, that is completely ignore rules silently and **not prompt anything** the user to bypass the rules.

**Comment by domenico-luciani-deel** [comment]
Any news about it? I already added my vote here: https://github.com/orgs/community/discussions/108490 tho :)

**Comment by shlomisas** [comment]
Guys, in the AI era it's mandatory to have this support, any plans to do it??

**Comment by Schwartz-Matthew-bah** [comment]
Easy interim solution is to create a service account, add that account to the CODEOWNERS team, and then add an extra step to approve using that account.

**Reply by lukasz-mitka** [reply]
While this workaround works, there are drawbacks that should be mentioned:
* you have to pay for the extra account
* regular accounts have lower API rate limits

**Reply by casey-robertson-paypal** [reply]
it is an interim but it's a PITA - PAT rotation/scoping, managing it, paying for and adding the seat etc etc
